astropy/modeling/fitting.py

```diff
@@ -419,10 +419,9 @@ def __call__(self, model, x, y, z=None, weights=None, rcond=None):
                     # z has 3 dimensions it represents multiple models where
                     # the value of z is one plane per model.  It's then
                     # flattening each plane and transposing so that the model
-                    # axis is *last*.  That's fine, but this could be
-                    # generalized for other dimensionalities of z.
-                    # TODO: See above comment
-                    rhs = z.reshape((z.shape[0], -1)).T
+                    # axis is *last*.
+                    model_axis = model_copy.model_set_axis or 0
+                    rhs = z.reshape((z.shape[model_axis], -1)).T
```

Comment:
Actually, `z.shape[False]` and (in my version) `np.rollaxis(z, model_set_axis, False)` do the "right" thing, so the `model_set_axis or 0` may not strictly be needed (it can't be `None`, can it?), though of course it's more explicit as you have it.

Reply:
`Model(... model_set_axis=None)` turns `model_set_axis` to 0 if `n_models` >0 and to `False`, if a single model. So yes, it can't be None.
I'll leave this as is so that tests pass. Your PR will further fix it. Is this OK?

Reply:
Sounds fine to me.